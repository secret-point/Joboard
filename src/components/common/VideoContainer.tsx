import React, { createRef, useState } from "react";

interface VideoContainerProps {
    src: string;
    poster?: string;
    type?: string;
    controls?: boolean;
    onClick?: Function;
    id: string;
}

const VideoContainer = ( props: VideoContainerProps ) => {
    const { id, controls, poster, src, type } = props;

    const [showControls, setShowControls] = useState<boolean>(false);
    const videoEl = createRef<HTMLVideoElement>();

    const onShowControl = ( e: React.MouseEvent<HTMLButtonElement> ) => {
        e.stopPropagation();
        e.nativeEvent.stopImmediatePropagation();
        //@ts-ignore
        videoEl.current.play();
        setShowControls(true);
    };

    return (
        <div className="video-container" data-testid="video">
            {!showControls && (
                <div className="videoPlayButton">
                    <button data-testid="play-button" onClick={onShowControl}>
                        <img
                            src="https://m.media-amazon.com/images/G/01/HVH-CandidateApplication/video_play_button.svg"></img>
                    </button>
                </div>
            )}
            <video
                data-testid={`video-${id}`}
                controls={controls || showControls}
                poster={poster}
                ref={videoEl}
                playsInline={true}
                preload="metadata"
            >
                <source src={`${src}#t=0.1`} type={type || "video/mp4"}/>
            </video>
        </div>
    );
};

export default VideoContainer;
