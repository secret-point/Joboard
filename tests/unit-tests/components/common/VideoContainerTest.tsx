import React from "react";
import { mount } from "enzyme";
import VideoContainer from "../../../../src/components/common/VideoContainer";
import { act } from "react-dom/test-utils";

describe("VideoContainer", () => {

    jest.spyOn(window.HTMLMediaElement.prototype, 'play').mockImplementation(async() => {});

    it("should match snapshot", () => {

        const shallowWrapper = mount(
          <VideoContainer
            src="https://v.amazon.com"
            poster="Test User"
            id="TEST-ID"
            onClick={jest.fn()}
          />);
        
        const controlButton = shallowWrapper.find('button[data-testid="play-button"]');

        act(()=>{
            controlButton.simulate('click', {
                stopPropagation: ()=>{},
                nativeEvent: {
                    stopImmediatePropagation: ()=>{}
                }
            });
        });
    
        expect(shallowWrapper).toMatchSnapshot();
    });
});