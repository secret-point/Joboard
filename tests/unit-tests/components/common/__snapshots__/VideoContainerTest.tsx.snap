// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VideoContainer should match snapshot 1`] = `
<VideoContainer
  id="TEST-ID"
  onClick={[MockFunction]}
  poster="Test User"
  src="https://v.amazon.com"
>
  <div
    className="video-container"
    data-testid="video"
  >
    <div
      className="videoPlayButton"
    >
      <button
        data-testid="play-button"
        onClick={[Function]}
      >
        <img
          src="https://m.media-amazon.com/images/G/01/HVH-CandidateApplication/video_play_button.svg"
        />
      </button>
    </div>
    <video
      controls={false}
      data-testid="video-TEST-ID"
      playsInline={true}
      poster="Test User"
      preload="metadata"
    >
      <source
        src="https://v.amazon.com#t=0.1"
        type="video/mp4"
      />
    </video>
  </div>
</VideoContainer>
`;
